session/session.go

```diff
@@ -0,0 +1,135 @@
+package session
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	"launchpad.net/gozk"
+)
+
+type ZKSessionEvent uint
+
+var ErrZKSessionNotConnected = errors.New("unable to connect to ZooKeeper")
+
+const (
+	SessionClosed ZKSessionEvent = iota
```

Comment:
Starting point for constant documentation:

```
const (
    // SessionClosed is normally only returned as a direct result of calling Close() on the ZKSession object. It is a terminal state; the connection will not be re-established.
    SessionClosed ZKSessionEvent = iota
    // SessionDisconnected is a transient state indicating that the connection to ZooKeeper was lost. The library is attempting to reconnect and you will receive another event when it has. In the meantime, if you're using ZooKeeper to implement, for example, a lock, assume you have lost the lock.
    SessionDisconnected
    // SessionReconnected is returned after a SessionDisconnected event, to indicate that the library was able to re-establish its connection to the zookeeper cluster before the session timed out. Ephemeral nodes have not been torn down, so any created by the previous connection still exist.
    SessionReconnected
    // SessionExpiredReconnected indicates that the session was reconnected (also happens strictly after a SessionDisconnected event), but that the reconnection took longer than the session timeout, and all ephemeral nodes were purged.
    SessionExpiredReconnected
    // SessionFailed indicates that the session failed unrecoverably. This may mean incorrect credentials, or broken quorum, or a partition from the entire zookeeper cluster, or any other mode of absolute failure.
    SessionFailed
)